File: solutions/free-pascal/hamming/1/Hamming.pas
unit Hamming;

{$mode ObjFPC}{$H+}

interface

function distance(const strand1 : string; const strand2 : string) : integer;

implementation

uses SysUtils;

function distance(const strand1 : string; const strand2 : string) : integer;
var
  i : integer;
begin
  if length(strand1)<>length(strand2) then
    raise Exception.Create('strands must be of equal length');
  result := 0;
  for i:=1 to length(strand1) do
  begin
    if strand1[i]<>strand2[i] then
      inc(result);      
  end;  

end;

end.

File: solutions/kotlin/hamming/1/src/main/kotlin/Hamming.kt
object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {
        require(leftStrand.length ==rightStrand.length) {
           "left and right strands must be of equal length"}
        var l=leftStrand.length
        var hdist = 0
        for(i in 0..(l-1)){
            if(leftStrand[i]!=rightStrand[i]){
                hdist ++; }
        }
        return hdist
    }
}
